nautobot/core/views/__init__.py

```diff
@@ -196,6 +197,20 @@ def server_error(request, template_name=ERROR_500_TEMPLATE_NAME):
     )
 
 
+def csrf_failure(request, reason="", template_name="403_csrf_failure.html"):
+    """Custom 403 CSRF failure handler to account for additional context.
+
+    If Nautobot is set to DEBUG the default view for csrf_failure.
+    `403_csrf_failure.html` template name is used over `403_csrf.html` to account for
+    additional context that is required to render the inherited templates.
+    """
+    if settings.DEBUG:
+        return _csrf_failure(request, reason=reason)
+    t = loader.get_template(template_name)
+    context = {"settings": settings}
+    return HttpResponseForbidden(t.render(context), content_type="text/html")
```

Comment:
`csrf_failure` accepts a `template_name` argument:

```suggestion
    return _csrf_failure(request, reason, template_name)
```

Reply:
But does not have our context, which we need settings for the template to render. Initially I just added the template which should have worked if it had access to the context. The upstream view does not allow for outside context.